test/es6/DeferParseLambda.js

```diff
@@ -110,6 +110,38 @@ var tests = [
             `);
         }
     },
+    {
+        name: "Global functions using 'yield' as identifier",
+        body: function () {
+            WScript.LoadScript(`
+                var a = async (yield) => { yield };
+                assert.isTrue(a() instanceof Promise, "Async lambda with yield as a formal parameter name");
+
+                function b(yield) {
+                    return yield;
+                }
+                assert.areEqual('b', b('b'), "Function with yield as a formal parameter name");
+
+                var c = async (yield) => yield;
+                assert.isTrue(c() instanceof Promise, "Async lambda with yield as a formal parameter name and compact body");
+            `);

```

Comment:
Doesn't need to be done in this PR, but I wish there was a way to make this test less fragile since there's no guarantee b would be deferred, right? I wonder if we need a WScript.SetFlags function that can force deferral.

Reply:
-force:deferparse should do the trick?

Reply:
In this case, probably. And given the name of the test, it probably already does that 😄 - I was more thinking aloud the general case of if we tests run with WScript.LoadScript which need a different set of flags to be reliable, whether we should have a mechanism to set those flags dynamically.

Reply:
Yes, we run this test file in three flavors with -force:deferparse, with -forceundodefer, and vanilla.